curator-x-discovery/src/main/java/org/apache/curator/x/discovery/details/ServiceCacheImpl.java

```diff
@@ -66,6 +68,18 @@
         cache.getListenable().addListener(this);
     }
 
+    ServiceCacheImpl(ServiceDiscoveryImpl<T> discovery, String name, CloseableExecutorService executorService)
```

Comment:
Please have the other ctor call this one. You may need to introduce a third private ctor. i.e. only 1 ctor should have code. The others should forward to the 1 with code.